src/libraries/System.Web.HttpUtility/src/System/Web/HttpUtility.cs

```diff
@@ -193,10 +193,33 @@ value switch
         public static byte[]? UrlDecodeToBytes(string? str) => UrlDecodeToBytes(str, Encoding.UTF8);
 
         [return: NotNullIfNotNull(nameof(str))]
-        public static byte[]? UrlDecodeToBytes(string? str, Encoding e) => str == null ? null : UrlDecodeToBytes(e.GetBytes(str));
+        public static byte[]? UrlDecodeToBytes(string? str, Encoding e)
+        {
+            if (str == null)
+            {
+                return null;
+            }
+
+            //do not execute Encoding.GetMaxByteCount if input is too big
```

Comment:
Style nit: spaces after `//` (same for other comments)
```suggestion
            // do not execute Encoding.GetMaxByteCount if input is too big
```

Reply:
@MihaZupan Do you think if this check is worth?
In general I am trying to skip Encoding.GetMaxByteCount, but if you say that this is fast enough and doesn't worth complicating code - I can remove it.

Reply:
`GetMaxByteCount` is cheap. I'd remove the branch